src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosJobFramework.scala

```diff
@@ -201,48 +201,52 @@ class MesosJobFramework @Inject()(
 
   @Override
   def statusUpdate(schedulerDriver: SchedulerDriver, taskStatus: TaskStatus) {
-    taskManager.taskCache.put(taskStatus.getTaskId.getValue, taskStatus.getState)
-
-    val (jobName, _, _, _) = TaskUtils.parseTaskId(taskStatus.getTaskId.getValue)
-    taskStatus.getState match {
-      case TaskState.TASK_RUNNING =>
-        scheduler.handleStartedTask(taskStatus)
-        updateRunningTask(jobName, taskStatus)
-      case TaskState.TASK_STAGING =>
-        scheduler.handleStartedTask(taskStatus)
-        updateRunningTask(jobName, taskStatus)
-      case _ =>
-        runningTasks.remove(jobName)
-    }
+    if(TaskUtils.isValidVersion(taskStatus.getTaskId.getValue)) {
+      val taskId = taskStatus.getTaskId.getValue
+      val state = taskStatus.getState
+      taskManager.taskCache.put(taskId, state)
+      val (jobName, _, _, _) = TaskUtils.parseTaskId(taskStatus.getTaskId.getValue)
```

Comment:
`TaskUtils.parseTaskId(taskStatus.getTaskId.getValue)`
could be 
`TaskUtils.parseTaskId(taskId)`